compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -2188,16 +2188,18 @@ pub(super) fn check_type_bounds<'tcx>(
                 //
                 // impl<T> X for T where T: X { type Y = <T as X>::Y; }
             }
-            _ => predicates.push(
-                ty::Binder::bind_with_vars(
+            _ => {
+                let pred: ty::Predicate<'_> = ty::Binder::bind_with_vars(
                     ty::ProjectionPredicate {
                         projection_ty: tcx.mk_alias_ty(trait_ty.def_id, rebased_args),
                         term: normalize_impl_ty.into(),
                     },
                     bound_vars,
                 )
-                .to_predicate(tcx),
-            ),
+                .to_predicate(tcx);
+
+                predicates.push(pred.expect_clause())
```

Comment:
This is a bit roundabout. We should probably add a way to create clauses directly

Reply:
yea, considering all the uses work like that, adding a `new` method to `Clause` seems best.

Reply:
I suppose it's something like this?
```
impl Clause {
   fn new(tcx: TyCtxt<'tcx>, pred: impl ToPredicate<'tcx>) -> Clause<'tcx> {
       // ...
   }
}
```

Reply:
just a `ClauseKind` as the argument. Even if internally you go through `ToPredicate`, there's nothing that can fail if the argument is `ClauseKind`

Reply:
Ohhh. nice.